src/apps/code-editor/src/app/components/PendingEditsModal/PendingEditsModal.js

```diff
@@ -0,0 +1,119 @@
+import { memo, useState, useEffect } from "react";
```

Comment:
I'm really confused on which component is the dirty code state modal. 

- src/shell/components/global-tabs/components/DirtyCodeModal.tsx
- src/apps/code-editor/src/app/components/PendingEditsModal/PendingEditsModal.js

These 2 components seem to be doing the same job but are loaded into the runtime at different places. What am I missing?

Reply:
They are logically different modals. I am in the process of refactoring so that they use the same component for *presentation,* but they are quite different logically:

`PendingEditsModal` is part of the Code app and is activated by the `getUserConfirmation` part of the `react-router` API whenever a user navigates away from the Code app. It operates on the code file the user is editing, getting this context from the URL path parts.

`DirtyCodeModal` exists globally. This can be triggered from anywhere in manager when a user unpins a tab. It operates on the URL that is saved in the pinned tab.